### DISCUSSION TITLE
When do upgrades get triggered?

### DISCUSSION THREAD

**Discussion by janpieper** [rancher/system-upgrade-controller#340] [Q&A]
Our k3s nodes get upgraded at aroud 3am in the morning, but I can't find any configuration that says that the upgrades should be triggered at that time.

Is there any documentation or explanation on when a system upgrade gets triggered? :thinking: The plans only contain information about the pre-conditions and what to do, but not when to do things.

**Comment by brandond** [comment]
By default the SUC polls the channel server periodically, and applies the update as soon as it is available.

Recent releases of the SUC have the option to put a time window on the plan: 
* https://github.com/rancher/system-upgrade-controller/pull/335
